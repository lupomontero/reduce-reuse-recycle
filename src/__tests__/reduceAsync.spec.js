const fs = require('fs');
const path = require('path');
// const { promisify } = require('util');
const reduceAsync = require('../reduceAsync');

describe('reduceAsync', () => {
  it('...', () => {
    const files = ['reduce.js', 'reverse.js'];
    // const readFile = promisify(fs.readFile);
    const readFile = (file, encoding) => Promise.resolve(`${file} ${encoding}...`);

    return reduceAsync(
      files.map(file => () => readFile(path.join(__dirname, '../', file), 'utf8')),
      (memo, result, idx) => ({ ...memo, [files[idx]]: result.length }),
      {},
    )
      .then(result => expect(result).toMatchSnapshot());
  });
});
