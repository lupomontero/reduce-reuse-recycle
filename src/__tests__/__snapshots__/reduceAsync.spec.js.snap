// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reduceAsync ... 1`] = `
Object {
  "reduce.js": 82,
  "reverse.js": 83,
}
`;
